core/domain/learner_playlist_services_test.py

```diff
@@ -179,6 +174,45 @@ def test_mark_exploration_to_be_played_later(self):
             self._get_all_learner_playlist_exp_ids(
                 self.user_id), exp_ids)
 
+        # Empty the learner playlist again.
+        for exp_id in exp_ids:
+            learner_playlist_services.remove_exploration_from_learner_playlist(
+                self.user_id, exp_id)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [])
+
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), [])
+
+        # Add MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT explorations at given
+        # positions.
+        for i in range(0, MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT):
+            learner_progress_services.add_exp_to_learner_playlist(
+                self.user_id, exp_ids[i], position_to_be_inserted=i)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(
+                self.user_id), exp_ids)
+
+        # Now if we try to add another exploration at any given position, it
+        # shouldn't be added as the list length would exceed
+        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
+        learner_playlist_services.mark_exploration_to_be_played_later(
+            self.user_id, 'SAMPLE_EXP_ID', position_to_be_inserted=1)
```

Comment:
Hmm, I am not sure here.... But this could also happen due to collision with an already existing element at position 1? Maybe the right thing to test is to try to add an element at the end of the list (like what would happen when a typical user tries to add an exploration after filling up all the spots). 

Actually, I think both of these would need to be tested. (Not sure if already tested in the file, if so could you mention the name of the test, or provide a link to it?). 

Let me know if you want to discuss this further.

Reply:
Hi Nithesh, thanks for the feedback, and sorry for the late response. 
I believe adding an element at an existing position is tested in lines 122-136. So, I left those lines unchanged. Varun's most recent submission on my behalf tests adding an element to the end of the list like you mention in this comment.